modules/ui/background.js

```diff
@@ -366,6 +371,17 @@ export function uiBackground(context) {
                 .content(renderOverlayList)
             );
 
+        // grid list
+        _gridOptionsContainer = content
+            .append('div')
+            .attr('class', 'grid-overlay-list-container'); 
+        
+        context.rapidContext().on('task_extent_set.background', function() {
+            if (!context.rapidContext().isTaskRectangular()) {
```

Comment:
isTaskRectangular() is pretty much only used to determine whether grid controls should be displayed at all. Should this be a different method name?

Reply:
I think it matches well to what it does. To help clarify the check, you can consider adding a comment like
`// only draw grids when task boundary is rectangular`